### ISSUE TITLE
Add scvi-tools as a Hugging Face library

### ISSUE THREAD

**Issue by adamgayoso** [huggingface/huggingface_hub#1129]
[scvi-tools](https://scvi-tools.org/) is a machine learning library for single-cell genomics data. It contains implementations of popular ML models in the field as well as abstractions for quickly developing new models. It has both Jax/PyTorch backends.

A primary task people use models in scvi-tools is for pre-training VAE-like models on large datasets and then fine-tuning on newly generated query data. An analysis might fine-tune query data and then operate on the latent space of the model (train + test embeddings together) in an unsupervised way. We have an example of this [here](https://docs.scvi-tools.org/en/stable/tutorials/notebooks/query_hlca_knn.html)

We wanted to add support for uploading to and downloading from Hugging Face in our library, which we have a good idea how to do. However, I wanted to check in to see if we can add scvi-tools as a library in Hugging Face.

Furthermore, I'm not sure how to classify this task based on Hugging Face tasks. Tabular classification makes sense in some contexts (data is tabular) but tabular representation learning is not there?

**Comment by osanseviero** [comment]
Hi @adamgayoso! 

Thanks for reaching out! Yes, you can add support for uploading and downloading model to/from the Hub. We have a guide on how to do that [here](https://huggingface.co/docs/hub/models-adding-libraries), including adding `scvi-tools` as an official supported library. 

In order to enable this as an official library, you can open a PR against [this file](https://github.com/huggingface/hub-docs/blob/main/js/src/lib/interfaces/Libraries.ts) in the hub-docs repository. Note though that we avoid adding official libraries until there are is a good number of models in the Hub. We do this to avoid having a filter in the Hub that leads to very few models. 

As for what task to use, let's discuss a bit more to make sure we do an appropriate name (cc @lhoestq @merveenoyan @julien-c )

**Comment by adamgayoso** [comment]
> In order to enable this as an official library, you can open a PR against [this file](https://github.com/huggingface/hub-docs/blob/main/js/src/lib/interfaces/Libraries.ts) in the hub-docs repository. Note though that we avoid adding official libraries until there are is a good number of models in the Hub. We do this to avoid having a filter in the Hub that leads to very few models.

Makes sense. We were going to upload on the order of 10+ models to start. Potentially naive question, but if the model library isn't there, can you change it after?

**Comment by osanseviero** [comment]
Yes! The library is automatically determined based on the model card metadata ([docs](https://huggingface.co/docs/hub/model-cards#model-card-metadata)). You can use either `tags` or [even better](https://huggingface.co/docs/hub/model-cards#specifying-a-library) you can use `library_name: scvi-tools` for example. Once that's done, you will be able to filter by library.

And once the PR is merged in `hub-docs`, it will appear in the left filter of hf.co/models and users will get an automatic code snippet as well
`

**Comment by merveenoyan** [comment]
@adamgayoso The names of the tasks are usually dependent on the abstractions over the input types, we can decide on this later as it requires a bit of more work 🙂 In the meantime, please let me know if you have any questions.

**Comment by Wauplin** [comment]
Hi @adamgayoso , I'm getting back to you about this issue :) What is the current state of the integration of `scvi-tool` as a Huggingface lib ?

**Comment by adamgayoso** [comment]
Hi @Wauplin we are getting ready to upload about a dozen models, I'll reach back out when we have completed this! cc @watiss

**Comment by Wauplin** [comment]
Hi @adamgayoso, glad that you made some progress to share the models!
I remain available if you need some help somewhere :)

**Comment by Wauplin** [comment]
Integrations of new libraries have changed since 2023. We now have an official guide for this: https://huggingface.co/docs/hub/models-adding-libraries. Will close this PR but remain available if needed
